ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/precompiles/privacy/PrivacyPrecompiledContract.java

```diff
@@ -206,17 +209,20 @@ void persistPrivateState(
         disposablePrivateState.rootHash(),
         privateStateUpdater);
 
-    final int txStatus =
-        result.getStatus() == PrivateTransactionProcessor.Result.Status.SUCCESSFUL ? 1 : 0;
+    maybeUpdateGroupHeadBlockMap(privacyGroupId, currentBlockHash, privateStateUpdater);
 
-    final PrivateTransactionReceipt privateTransactionReceipt =
-        new PrivateTransactionReceipt(
-            txStatus, result.getLogs(), result.getOutput(), result.getRevertReason());
+    processTransactionReceipt(commitmentHash, currentBlockHash, result, privateStateUpdater);
+  }
 
-    privateStateUpdater.putTransactionReceipt(
-        currentBlockHash, commitmentHash, privateTransactionReceipt);
+  void processTransactionReceipt(
+      final Hash pmtHash,
+      final Hash currentBlockHash,
+      final PrivateTransactionProcessor.Result result,
+      final PrivateStateStorage.Updater privateStateUpdater) {
+    final PrivateTransactionReceipt privateTransactionReceipt =
+        new PrivateTransactionReceipt(result);
 
-    maybeUpdateGroupHeadBlockMap(privacyGroupId, currentBlockHash, privateStateUpdater);
+    privateStateUpdater.putTransactionReceipt(currentBlockHash, pmtHash, privateTransactionReceipt);
 
     privateStateUpdater.commit();
```

Comment:
I think the code would be easier to read if you moved the commit out of this method, as there is data being committed that is not created in this method.